sopel/tools/target.py

```diff
@@ -107,9 +108,193 @@ def add_user(self, user, privs=0):
         """
         assert isinstance(user, User)
         self.users[user.nick] = user
-        self.privileges[user.nick] = privs
+        self.privileges[user.nick] = privs or 0
         user.channels[self.name] = self
 
+    def has_privilege(self, nick, privilege):
+        """Tell if a user has a ``privilege`` level or above in this channel.
+
+        :param str nick: a user's nick in this channel
+        :param int privilege: privilege level to check
+        :rtype: bool
+
+        This method checks the user's privilege level in this channel, i.e. if
+        it has this level or higher privileges::
+
+            >>> channel.add_user(some_user, plugin.OP)
+            >>> channel.has_privilege(some_user.nick, plugin.VOICE)
+            True
+
+        The ``nick`` argument can be either a :class:`str` or a
+        :class:`sopel.tools.Identifier`. If the user is not in this channel,
+        it will be considered has not having any privilege.
```

Comment:
```suggestion
        it will be considered as not having any privilege.
```

Must have missed this one before. Oops.